cpp/include/raft/distance/detail/pairwise_distance_cutlass_base.cuh

```diff
@@ -59,26 +62,29 @@ template <typename DataT,
           typename IdxT,
           int VecLen,
           typename FinalLambda,
-          typename DistanceFn,
+          typename OpT,
           bool isRowMajor>
-void cutlassDistanceKernel(const DataT* x,
-                           const DataT* y,
-                           const DataT* xn,
-                           const DataT* yn,
-                           IdxT m,
-                           IdxT n,
-                           IdxT k,
-                           IdxT lda,
-                           IdxT ldb,
-                           IdxT ldd,
-                           OutT* dOutput,
-                           FinalLambda fin_op,
-                           DistanceFn dist_op,
-                           cudaStream_t stream)
+typename std::enable_if<ops::has_cutlass_op<OpT>::value>::type cutlassDistanceKernel(
+  const DataT* x,
+  const DataT* y,
+  const DataT* xn,
+  const DataT* yn,
+  IdxT m,
+  IdxT n,
+  IdxT k,
+  IdxT lda,
+  IdxT ldb,
+  IdxT ldd,
+  OutT* dOutput,
+  FinalLambda fin_op,
+  OpT distance_op,
+  cudaStream_t stream)
 {
   static_assert(!(std::is_same<OutT, bool>::value),
                 "OutType bool is not supported use uint8_t instead");
 
+  auto dist_op     = distance_op.get_cutlass_op();
```

Comment:
```suggestion
  auto constexpr dist_op     = distance_op.get_cutlass_op();
```

Reply:
This will not work: `distance_op` is a function parameter. In the case of L2 expanded, it is *not* a zero-sized struct (as it contains the runtime boolean `sqrt`). Therefore `distance_op` is not constexpr and the code will not compile. 

At first, this change did compile for me. The reason was that I had removed the ampersand in `struct has_cutlass_op<T, std::void_t<decltype(&T::get_cutlass_op)>> : std::true_type {};`. As a result, the compilation of the cutlass kernels was skipped. 

When I added back the ampersand, the cutlass kernels were compiled again and this change caused a compiler error.